src/Credentials/CredentialProvider.php

```diff
@@ -581,6 +581,33 @@ private static function getHomeDir()
         return ($homeDrive && $homePath) ? $homeDrive . $homePath : null;
     }
 
+    /**
+     * Gets profiles from specified $filename, or default ini files.
+     */
+    private static function loadProfiles($filename)
+    {
+        $profileData = \Aws\parse_ini_file($filename, true, INI_SCANNER_RAW);
+
+        // If loading .aws/credentials, also load .aws/config when AWS_SDK_LOAD_NONDEFAULT_CONFIG is set
+        if ($filename === self::getHomeDir() . '/.aws/credentials'
+            && getenv('AWS_SDK_LOAD_NONDEFAULT_CONFIG')
+        )
+        {
+            $configFilename = self::getHomeDir() . '/.aws/config';
+            $configProfileData = \Aws\parse_ini_file($configFilename, true, INI_SCANNER_RAW);
+            foreach ($configProfileData as $name => $profile)
+            {
+                // standardize config profile names
```

Comment:
Does this behavior match other SDKs/CLI?

Reply:
When loading from config, JavaScript strips the 'profile ' prefix:
https://github.com/aws/aws-sdk-js/blob/9b64887b9f6d12b8f87389724aa31bb90651573f/lib/shared-ini/ini-loader.js#L10

And then merges all of the credential and config profiles into one object.  config profiles get overwritten by credential profiles if they have the same name.